main/src/main/scala/sbt/internal/LintUnused.scala

```diff
@@ -165,6 +165,39 @@ object LintUnused {
     unusedKeys.map(u => (u.scoped, display.show(u.scoped), u.positions)).sortBy(_._2)
   }
 
+  def lintScalaVersion(state: State): State = {
+    val log = state.log
+    val extracted = Project.extract(state)
+    val structure = extracted.structure
+    val comp = structure.compiledMap
+    for
+      p <- structure.allProjectRefs
+      scope = Scope.Global.rescope(p)
+      key = scalaVersion.rescope(scope)
+      data = Project.scopedKeyData(structure, key.scopedKey)
+      sv = extracted.get(key)
+      isPlugin = extracted.get(sbtPlugin.rescope(scope))
+      mb = extracted.get(isMetaBuild.rescope(scope))
+      auto = extracted.get(autoScalaLibrary.rescope(scope))
+      (_, sk) = extracted.runTask(skip.rescope(scope.rescope(publish.key)), state)
+      display = p match
+        case ProjectRef(_, id) => id
+        case _ | null          => Reference.display(p)
+      c <- comp.get(data.map(_.definingKey).getOrElse(key.scopedKey))
+      setting <- c.settings.headOption
+    do
+      if auto && !isPlugin && !mb && !sk then
```

Comment:
Hey @eed3si9n, does this work for

```
sbt:colors> autoScalaLibrary
[info] true
sbt:colors> managedScalaInstance
[info] false
```

or would the warning be intended in that case?

Reply:
Good call. I can exempt `managedScalaInstance` as well.